src/in-depth/machine-communication.md

```diff
@@ -230,10 +230,58 @@ This is how Visual Studio Code uses _ripgrep_ for its code search.
 
 ## How to deal with input piped into us
 
-<aside class="todo">
 
-**TODO:**
-Talk about how work with stdin
-(see [#95](https://github.com/rust-cli/meta/issues/95))
+Rust programs can read data passed in via `stdin`, whether that's through
+piping or text entered while the program is running. Rust's standard library
+provides us with
+[`std::io::stdin`](https://doc.rust-lang.org/std/io/fn.stdin.html) that can
+read the lines passed in.
 
-</aside>
+Here's a program that counts the words in all of the lines of what's passed in
+via `stdin`. Note, the `-` arg for accepting input from stdin comes from the [Command Line Interface Guidelines](https://clig.dev), a helpful resource when building CLIs.
+
+``` rust,ignore
+use std::io::stdin;
+
+fn main() {
+    let mut total_word_count = 0;
+
+    let error = "pass in - to read from stdin";
+    let file = std::env::args().nth(1).expect(error);
+
+    if file == "-" {
```

Comment:
Don't we also want to default to `-` if stdin is open and error if not?

Reply:
I'm not exactly following what you're suggesting here. In the revised code though, I made no file display the help text and `-` is required to read from stdin, whether piped in or interactive. Does that cover this?

Reply:
I did some investigation into how to determine if stdin is or isn't a tty, and it seems like the [atty](https://crates.io/crates/atty) crate a cross-platform way to handle this. Would it be appropriate to use it in the example code for this book?

Reply:
I went with atty since it was simple enough to add and other crates are used in the project. Let me know if you want that approach changed.